libs/core/src/lib/action-sheet/action-sheet.component.ts

```diff
@@ -0,0 +1,231 @@
+import {
+    AfterContentInit,
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    ContentChild,
+    ViewChild,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Injector,
+    Input,
+    OnDestroy,
+    Optional,
+    Output,
+    ViewEncapsulation,
+    TemplateRef,
+    ComponentRef,
+    QueryList,
+    ContentChildren
+} from '@angular/core';
+import { PopoverComponent } from '../popover/popover.component';
+import { DynamicComponentService } from '../utils/dynamic-component/dynamic-component.service';
+import { ActionSheetBodyComponent } from './action-sheet-body/action-sheet-body.component';
+import { ActionSheetControlComponent } from './action-sheet-control/action-sheet-control.component';
+import {
+    FocusEscapeDirection,
+    KeyboardSupportService
+} from '../utils/services/keyboard-support/keyboard-support.service';
+import { ActionSheetItemComponent } from './action-sheet-item/action-sheet-item.component';
+import { startWith, takeUntil } from 'rxjs/operators';
+import {merge, Observable, Subject} from 'rxjs';
+import { ActionSheetMobileComponent } from './action-sheet-mobile/action-sheet-mobile.component';
+import {ACTION_SHEET_COMPONENT, ActionSheetInterface} from './action-sheet.interface';
+
+@Component({
+    selector: 'fd-action-sheet',
+    templateUrl: './action-sheet.component.html',
+    styleUrls: ['./action-sheet.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [
+        KeyboardSupportService
+    ]
+})
+export class ActionSheetComponent implements AfterContentInit, AfterViewInit, OnDestroy, ActionSheetInterface {
+
+    /** Whether should be displayed in compact mode **/
+    @Input()
+    compact = false;
+
+    /** Whether should be displayed in mobile mode **/
+    @Input()
+    mobile = false;
+
+    /** Whenever links should be visible **/
+    @Input()
+    open = false;
+
+    /** Whether internal keyboard support should be enabled. It's enabled by default */
+    @Input()
+    keyboardSupport = true;
+
+    /** To allow user to determine what event he wants to trigger the messages to show
+     * Accepts any [HTML DOM Events](https://www.w3schools.com/jsref/dom_obj_event.asp).
+     **/
+    @Input()
+    triggers: string[] = ['click'];
+
+    /** Event thrown, when focus escapes the list */
+    @Output()
+    focusEscapeList = new EventEmitter<FocusEscapeDirection>();
+
+    /** Event thrown, when the action sheet is opened or closed */
+    @Output()
+    readonly openChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    /** @hidden **/
+    @Input()
+    refreshObs$: Observable<string>;
```

Comment:
Why is it @Input()?
Also it should be `Observable<string>` type.
Consider removing it, because I can't see any usage of this property